### PR TITLE
feat: handler pattern for starting  new processes

### PR DESCRIPTION
**Pull request by MadsDue**
## Description
Prepare our Process module to handle multiple Initialize process types.

### CODE REVIEW

File: source/Api/EventListeners/ProcessInitializationListener.cs [hunk 1]
```diff
@@ -26,30 +24,29 @@ namespace Energinet.DataHub.EDI.Api.EventListeners;
...
     }
 
     [Function(nameof(ProcessInitializationListener))]
     public async Task RunAsync(
         [ServiceBusTrigger(
             "%INCOMING_MESSAGES_QUEUE_NAME%",
             Connection = "SERVICE_BUS_CONNECTION_STRING_FOR_DOMAIN_RELAY_LISTENER")]
-        byte[] eventData,
+        ServiceBusReceivedMessage message,
         FunctionContext context)
```
Comment (RasmusGraabaek) [inline source/Api/EventListeners/ProcessInitializationListener.cs:43]: I do believe that we can remove the context, since the message is typed

**Review by RasmusGraabaek — APPROVED** [review]

File: source/Process.Interfaces/IProcessInitializationClient.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// limitations under the License.
+
+using System.Threading.Tasks;
+
+namespace Energinet.DataHub.EDI.Process.Interfaces;
+
+/// <summary>
+/// Client for initializing processes.
+/// </summary>
+public interface IProcessInitializationClient
```
Comment (ebbeknudsen) [inline source/Process.Interfaces/IProcessInitializationClient.cs:22]: Shouldn't this just be `IProcessClient` to follow the naming we use for our other clients?

**Review by ebbeknudsen — DISMISSED** [review]

File: source/Process.Interfaces/IProcessInitializationClient.cs (hunk 2 above)
Comment (MadsDue) [reply source/Process.Interfaces/IProcessInitializationClient.cs:22]: Yes, you are rigth :D but currently, it is only used for initializing processes. But lets have a chat about it tomorrow :D

File: source/Api/EventListeners/ProcessInitializationListener.cs [hunk 3]
```diff
@@ -26,30 +24,29 @@ namespace Energinet.DataHub.EDI.Api.EventListeners;
...
     {
+        ArgumentNullException.ThrowIfNull(message);
         ArgumentNullException.ThrowIfNull(context);
         var eventDetails = context.ExtractEventDetails();
```
Comment (RasmusGraabaek) [inline source/Api/EventListeners/ProcessInitializationListener.cs:47]: When we hit this, the messageId of the service bus message does not exist. Hence the method "context.ExtractEventDetails();" til return an empty eventdetails. Giving us a useless log statement.

I would therefore prefer that we remove this loggin and might aswell kill the context from our trigger.

**Review by RasmusGraabaek — APPROVED** [review]

**Review by RasmusGraabaek — APPROVED** [review]

**Pull request merged**